exchange/bidder_test.go

```diff
@@ -2228,6 +2228,32 @@ func TestTimeoutNotificationOn(t *testing.T) {
 	assert.EqualValues(t, logExpected, logActual)
 }
 
+func TestTmaxInDoRequest(t *testing.T) {
+	ctx, cancelFunc := context.WithDeadline(context.Background(), time.Now().Add(50*time.Second))
+	defer cancelFunc()
+
+	bidRequest := adapters.RequestData{
+		Method: "POST",
+		Uri:    "test.url.com",
+		Body:   []byte(`{"id":"this-id","app":{"publisher":{"id":"pub-id"}}}`),
+	}
+
+	bidderAdapter := bidderAdapter{
+		me: &metricsConfig.NilMetricsEngine{},
+	}
+	logger := func(msg string, args ...interface{}) {}
+
+	httpCallInfo := bidderAdapter.doRequestImpl(ctx, &bidRequest, logger, time.Now(), &config.TmaxAdjustments{Enabled: true, BidderLatencyAdjustment: 20000, BidderResponseMin: 40000})
+	assert.NotNil(t, httpCallInfo.err)
+	assert.Equal(t, tmaxTimeoutErr, httpCallInfo.err)
+
+	httpCallInfo = bidderAdapter.doRequestImpl(ctx, &bidRequest, logger, time.Now(), &config.TmaxAdjustments{Enabled: true, BidderLatencyAdjustment: 20000, BidderResponseMin: 10000})
+	assert.NotEqual(t, tmaxTimeoutErr, httpCallInfo.err)
+
+	httpCallInfo = bidderAdapter.doRequestImpl(ctx, &bidRequest, logger, time.Now(), &config.TmaxAdjustments{Enabled: false, BidderLatencyAdjustment: 20000, BidderResponseMin: 40000})
+	assert.NotEqual(t, tmaxTimeoutErr, httpCallInfo.err)
+}
```

Comment:
This looks like 3 test cases. Consider either using t.Run for each or a table driven test design.

Reply:
https://github.com/prebid/prebid-server/pull/2787/commits/cb18ddff6ee148d7cba9a783cbb6b486cdd889f8